coordinator/src/db/funding_fee_events.rs

```diff
@@ -0,0 +1,142 @@
+use crate::funding_fee;
+use crate::schema::funding_fee_events;
+use crate::schema::protocol_funding_fee_events;
+use bitcoin::secp256k1::PublicKey;
+use bitcoin::SignedAmount;
+use diesel::prelude::*;
+use std::str::FromStr;
+use time::OffsetDateTime;
+use xxi_node::node::ProtocolId;
+
+#[derive(Queryable, Debug)]
+struct FundingFeeEvent {
+    id: i32,
+    /// A positive amount indicates that the trader pays the coordinator; a negative amount
+    /// indicates that the coordinator pays the trader.
+    amount_sats: i64,
+    trader_pubkey: String,
+    position_id: i32,
+    due_date: OffsetDateTime,
+    paid_date: Option<OffsetDateTime>,
+    #[diesel(column_name = "timestamp")]
+    _timestamp: OffsetDateTime,
+}
+
+pub(crate) fn insert(
+    conn: &mut PgConnection,
+    amount: SignedAmount,
+    trader_pubkey: PublicKey,
+    position_id: i32,
+    due_date: OffsetDateTime,
+) -> QueryResult<()> {
+    let res = diesel::insert_into(funding_fee_events::table)
+        .values(&(
+            funding_fee_events::amount_sats.eq(amount.to_sat()),
+            funding_fee_events::trader_pubkey.eq(trader_pubkey.to_string()),
+            funding_fee_events::position_id.eq(position_id),
+            funding_fee_events::due_date.eq(due_date),
+        ))
+        .execute(conn);
+
+    match res {
+        Ok(0) => Err(diesel::result::Error::NotFound),
+        Err(diesel::result::Error::DatabaseError(
+            diesel::result::DatabaseErrorKind::UniqueViolation,
+            _,
+        )) => {
+            tracing::debug!(
```

Comment:
Is this expected to happen? If not, shall we set this to `warn`?

Reply:
This can happen if we run the task too often and we try to insert the same funding fee event more than once. There is no harm if this does happen, so I think `debug` is fine.